src/Sextant/Navigation/ViewStackServiceBase.cs

```diff
@@ -224,15 +224,20 @@ protected static void PopRootAndTick<T>(BehaviorSubject<IImmutableList<T>> stack
             }
 
             stackSubject
-               .Take(stackSubject.Value.Count - 1)
+               .Take(1)
                .Where(stack => stack != null)
                .Subscribe(stack =>
                {
-                   poppedStack = stack.RemoveRange(stack.IndexOf(stack[0]), stack.Count - 1);
+                   if (stack.Count > 1)
+                   {
+                       poppedStack = stack.RemoveRange(stack.IndexOf(stack[1]), stack.Count - 1);
+                   }
+                   else
+                   {
+                       poppedStack = stack;
+                   }
                });
 
-            // var stack = stackSubject.Value;
-            // stack = stack.RemoveRange(stack.IndexOf(stack[0]), stack.Count - 1);
             stackSubject.OnNext(poppedStack);
```

Comment:
I wasn't certain because I can't get my Android emulator to hit breakpoints during debgging inside the Sextant assembly... but I think your code had a bug here.  The purpose of this section is to clear the stack after a PopToRoot (and all the navigation has been finished) and leave behind the root page's ViewModel.  However, your code seemed to be removing the first through the second to last viewmodels and only keeping the last one.

I probably won't ever matter on a Xamarin.Forms project because the initial instance still has its ViewModel, but when I was trying to pull the ViewModel from this service (no longer doing it), I kept getting the wrong view model.  I corrected it but I didn't simplify it.

Reply:
Report the bug in a separate issue and we'll take a look.  I think what your facing is Xamarin.Forms technically always has to have a page on the stack, so the stack always has to have a view model associated to whatever page is there.  I am interested in cleaning this up if there is a bug.